feat: adding enrichments (#186)
Basis-Theory/developers.basistheory.com: docs/api/tokens/token-enrichments.mdx

| ------------- | ----- | ---------------------------------------------------------------------------------------------- | ------------------------------------------------------------------ |
| `bin_details` | _any_ | [`card`](docs/api/tokens/token-types#card) , [`card_number`](docs/api/tokens/token-types#card) | Object containing information related to the tokenized credit card |

## Using Token Enrichments

When a specific Enrichment is activated for a Tenant, each is added as a token property and returned during every [token operation](/docs/api/tokens/) and [Detokenization Expressions](/docs/expressions/detokenization#using-token-properties).

## BIN Details <Enterprise />

BIN details, or Bank Identification Number details, represent a fundamental aspect of credit card processing.
Each credit card is associated with a unique BIN, which serves as a numerical identifier for the issuing bank
or financial institution. These details hold crucial technical significance as they help validate transactions,
prevent fraud, and optimize routing. Understanding the intricate mechanics of BIN details
is vital for implementing secure and efficient credit card processing solutions.

> In this section of Basis Theory's documentation, we outline the options you have when interacting with
> BIN details within the Basis Theory platform and how our tokenization services
> seamlessly integrate this crucial element to enhance data security and compliance in your applications.
> 
> Basis Theory offers two options for BIN details:

Reviewer: ```suggestion
Basis Theory offers two options for enriching `card` or `card_number` tokens with BIN details, depending on the level of detail you require:
```

| Level                      | Description                                                                                  |
| -------------------------- | -------------------------------------------------------------------------------------------- |

> | [Standard](#standard-data) | Details on the credit cards function and origin.                                             |

Reviewer: ```suggestion
| [Standard](#standard-data) | Basic information about the credit card's function and origin.                                             |
```

| Property            | Type   | Description                                  |
| ------------------- | ------ | -------------------------------------------- |
| bank                | object |                                              |
| _&nbsp;.name_       | string | Issuing bank name                            |
| _&nbsp;.phone_      | string | Issuing bank phone                           |
| _&nbsp;.url_        | string | Issuing bank website                         |
| _&nbsp;.clean_name_ | string | Issuing bank name                            |
| card_brand          | string | Card brand                                   |
| card_segment_type   | string | Consumer BIN or commercial BIN               |
| country             | object |                                              |
| _&nbsp;.alpha2_     | string | Issuing country ISO3166 aplpha country code  |
| _&nbsp;.numeric_    | string | Issuing country ISO3166 numeric country code |
| _&nbsp;.name_       | string | Issuing country name                         |
| prepaid             | bool   | Is prepaid BIN                               |
| product             | object |                                              |

> | _&nbsp;.code_       | string | The product ID                               |

Reviewer: ```suggestion
| _&nbsp;.code_       | string | The product code                               |
```
